src/Tmds.Systemd/ServiceManager.cs

```diff
@@ -33,5 +37,57 @@ public static string InvocationId
                 return _invocationId;
             }
         }
+
+        private static bool CheckServiceManager()
+        {
+            // No point in testing anything unless it's Unix
+            if (Environment.OSVersion.Platform != PlatformID.Unix)
+            {
+                return false;
+            }
+
+            // We've got invocation id, it's systemd >= 232 running a unit
+            if (InvocationId != null)
```

Comment:
We can leave this check out, right?

Reply:
Probably yes, I just didn't want to change anything for newer versions of systemd. I don't want to break everything for my special needs.

Reply:
I think of this PR more of a compat layer, not the primary "improved" implementation, so I decided to change as little as possible when it can be done.

Reply:
ok, let's keep it as it was. That makes non-direct children also return true with recent systemd versions (unless INVOCATION_ID is cleared).

Reply:
I thought about going up the process hierarchy when detecting systemd, but thought that it would be too complex for what I am trying to accomplish.

Reply:
I was thinking about that too, until I realized that the root of the tree is systemd :smile: